pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -27,8 +35,13 @@
                        hostPlatform != stdenv.buildPlatform
 , extraMeta ? {}
 , hostPlatform
+
+# easy overrides to hostPlatform.platform members
+, autoModules ? hostPlatform.platform.kernelAutoModules
+, preferBuiltin ? if (hostPlatform.platform ? kernelPreferBuiltin) then hostPlatform.platform.kernelPreferBuiltin else false
```

Comment:
````
, preferBuiltin ? hostPlatform.platform.kernelPreferBuiltin or false
````

Reply:
ok didn't know. Is the `or` specific to `A ? B or C` structure ? tried to use it in other places without success.

Reply:
It's related to the attribute set syntax: `X.Y or Z` is basically the same as `if X ? Y then X.Y else Z`.